components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -394,9 +394,13 @@ public List<APIRuntimeArtifactDto> retrieveGatewayArtifactsByAPIIDAndLabel(Strin
     public List<APIRuntimeArtifactDto> retrieveGatewayArtifactsByAPIIDs(List<String> apiIds, String[] labels,
                                                                         String tenantDomain)
             throws APIManagementException {
+        int apiIdListSize = apiIds.size();
+        // Logging the API ID List
+        for (int i = 0; i < apiIdListSize; i++) {
+            log.info("Getting runtime artifacts for the API ID: " + apiIds.get(i));
```

Comment:
shall we make this a debug log..
Shall we add all IDs in a single line using a string builder?

Reply:
Or since it's just a list of strings, wouldn't simply calling its `toString()` do?

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/11229/commits/c9cdba69f446b8698d33714d7ae49fdb3947ce33